File: sdk/fsharp/src/HiTechCloudSDK/Hosting.fs
namespace HiTechCloud.SDK

/// Hosting Services: reinstall, status, bandwidth, reset
type HostingResource(httpClient: HttpClient) =

    /// GET /api/service/{id}/reinstall - Return installation options
    member this.GetReinstallDetails() =
        async {
            return! httpClient.getAsync("/api/service/{id}/reinstall")
        }

    /// POST /api/service/{id}/reinstall - Reinstall server
    member this.Reinstall() =
        async {
            return! httpClient.postAsync("/api/service/{id}/reinstall")
        }

    /// GET /api/service/{id}/status - Return server power status
    member this.GetStatus() =
        async {
            return! httpClient.getAsync("/api/service/{id}/status")
        }

    /// GET /api/service/{id}/bandwidth-graphs - Return bandwidth usage graphs
    member this.GetBandwidthGraphs() =
        async {
            return! httpClient.getAsync("/api/service/{id}/bandwidth-graphs")
        }

    /// GET /api/service/{id}/bandwidth - Return bandwidth usage and billing
    member this.GetBandwidthUsage() =
        async {
            return! httpClient.getAsync("/api/service/{id}/bandwidth")
        }

    /// POST /api/service/{id}/reset - Reset server
    member this.Reset() =
        async {
            return! httpClient.postAsync("/api/service/{id}/reset")
        }

File: sdk/clojure/src/hitechcloud/sdk/hosting.clj
(ns hitechcloud.sdk.hosting
  (:require [hitechcloud.sdk.http-client :as http]))

;; Hosting Services: reinstall, status, bandwidth, reset

(defn get_reinstall_details "GET /api/service/{id}/reinstall - Return installation options" [client & [params]] (http-get client "/api/service/{id}/reinstall" params))

(defn reinstall "POST /api/service/{id}/reinstall - Reinstall server" [client & [params]] (http-post client "/api/service/{id}/reinstall" params))

(defn get_status "GET /api/service/{id}/status - Return server power status" [client & [params]] (http-get client "/api/service/{id}/status" params))

(defn get_bandwidth_graphs "GET /api/service/{id}/bandwidth-graphs - Return bandwidth usage graphs" [client & [params]] (http-get client "/api/service/{id}/bandwidth-graphs" params))

(defn get_bandwidth_usage "GET /api/service/{id}/bandwidth - Return bandwidth usage and billing" [client & [params]] (http-get client "/api/service/{id}/bandwidth" params))

(defn reset "POST /api/service/{id}/reset - Reset server" [client & [params]] (http-post client "/api/service/{id}/reset" params))

